js/ui/iconGrid.js

```diff
@@ -317,7 +317,7 @@ const BaseIcon = new Lang.Class({
             this.createIcon = params.createIcon;
         this._setSizeManually = params.setSizeManually;
 
-        this.icon = null;
+        this.icons = [];
```

Comment:
Yes, this is a better idea. Added the `createExtraIcons()` method just for that.